fivex/api/format.py

```diff
@@ -290,84 +337,208 @@ def pvalue(self):
+            skip_rows=rowstoskip
+        )
 
-    def __call__(self, variant: VariantContainer) -> VariantContainer:
-        # Cluster is a numeric index for a group of variants in LD in the DAP-G model, and are specific to a gene
-        # PIP is the Posterior Inclusion Probability for a single variant
-        # SPIP is the sum of PIPs for all variants belonging to the same cluster
+        # If the query is single variant, set end to (start + 1) 
+        # and use the trick from query_variants to get a single position
+        if end is None:
+            reader.add_filter("position", start)
+            ciRows = reader.fetch(chrom, start - 1, start + 1)
+        # If the query is regional, then filter for the gene of interest
+        # and get the study-, tissue-, and gene-specific data from the entire region
+        else:
+            reader.add_filter("gene_id", gene_id)
+            ciRows = reader.fetch(chrom, start - 1, end + 1)
+        for row in ciRows:
+            key = ":".join(
+                [
+                    row.chromosome,
+                    str(row.position),
+                    row.ref_allele,
+                    row.alt_allele,
+                    row.study,
+                    row.tissue,
+                    row.gene_id,
+                ]
+            )
+            # Dictionary Format: ci_Dict[chrom:pos:ref:alt:study:tissue:gene_id] = (cs_index, cs_size, pip)
+            ci_data[key] = (row.cs_index, row.cs_size, row.pip)
+        self.ci_data = ci_data
 
-        default = (0, 0.0, 0.0)
-        if self.pip_data is None:
-            (cluster, spip, pip) = default
+    def __call__(self, variant: VariantContainer) -> VariantContainer:
+        default = ("-", 0, 0.0)
+        if self.ci_data is None:
+            # cs_index is our new cluster (L1 or L2); we will repurpose spip with cs_size for the size of the cluster
+            (cs_index, cs_size, pip) = default
         else:
-            # In the PIP dictionary, chrom:pos:ref:alt:tissue:gene (with version number) is used as the Python dict key,
-            #  and (cluster, spip, pip) is the value
-            (cluster, spip, pip) = self.pip_data.get(
+            (cs_index, cs_size, pip) = self.ci_data.get(
                 ":".join(
                     [
-                        "chr" + variant.chromosome,
+                        variant.chromosome,
                         str(variant.position),
                         variant.ref_allele,
                         variant.alt_allele,
+                        variant.study,
                         variant.tissue,
                         variant.gene_id,
                     ]
                 ),
                 default,  # Some variants may lack information
             )
-
-        variant.pip_cluster = cluster
-        variant.spip = spip
+        variant.cs_index = cs_index
+        variant.cs_size = cs_size
         variant.pip = pip
         return variant
 
 
+# class PipAdder:
```

Comment:
Reminder to self: before merge, we should try to remove this code instead of just commenting it out.